cucumber-core/src/test/java/io/cucumber/core/plugin/HtmlFormatterTest.java

```diff
@@ -86,11 +89,17 @@ void ignores_step_definitions() throws Throwable {
             null);
         bus.send(Envelope.of(testRunFinished));
 
-        assertThat(bytes, bytes(containsString("" +
-                "window.CUCUMBER_MESSAGES = [" +
-                "{\"testRunStarted\":{\"timestamp\":{\"seconds\":10,\"nanos\":0}}}," +
-                "{\"testRunFinished\":{\"success\":true,\"timestamp\":{\"seconds\":15,\"nanos\":0}}}" +
-                "];\n")));
+        assertEquals("[" +
+                "{\"testRunStarted\":{\"timestamp\":{\"nanos\":0,\"seconds\":10}}}," +
+                "{\"testRunFinished\":{\"success\":true,\"timestamp\":{\"nanos\":0,\"seconds\":15}}}" +
+                "]",
+            extractCucumberMessages(bytes), STRICT);
     }
 
+    private static String extractCucumberMessages(ByteArrayOutputStream bytes) {
+        Pattern pattern = Pattern.compile("^.*window\\.CUCUMBER_MESSAGES = (\\[.+]);.*$", Pattern.DOTALL);
```

Comment:
The `Pattern` should compiled only once, so should be as a constant in the class.

Reply:
We'll revisit that once there are more than two testcases. :smile: